src/Api/Request/ParametersBuilder.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Findologic\Api\Request;
+
+use Findologic\Constants\Plugin;
+use Plenty\Log\Contracts\LoggerContract;
+use Plenty\Modules\Category\Models\Category;
+use Plenty\Plugin\Http\Request as HttpRequest;
+use Plenty\Plugin\Log\LoggerFactory;
+use IO\Services\CategoryService;
+
+class ParametersBuilder
+{
+    /**
+     * @var CategoryService
+     */
+    protected $categoryService;
+
+    /**
+     * @var LoggerContract
+     */
+    protected $logger;
+
+    public function __construct( LoggerFactory $loggerFactory)
+    {
+        $this->logger = $loggerFactory->getLogger(Plugin::PLUGIN_NAMESPACE, Plugin::PLUGIN_IDENTIFIER);
+    }
+
+    /**
+     * @return CategoryService|null
+     */
+    public function getCategoryService()
+    {
+        if (!$this->categoryService) {
+            $this->categoryService = pluginApp(CategoryService::class);
+        }
+
+        return $this->categoryService;
+    }
+
+    /**
+     * @param Request $request
+     * @param HttpRequest $httpRequest
+     * @param Category|null $category
+     * @return Request
+     */
+    public function setSearchParams($request, $httpRequest, $category = null)
+    {
+        $parameters = $httpRequest->all();
+
+        $request->setParam('query', $parameters['query'] ?? '');
+        $request->setPropertyParam(Plugin::API_PROPERTY_MAIN_VARIATION_ID);
+
+        if (isset($parameters[Plugin::API_PARAMETER_ATTRIBUTES])) {
+            $attributes = $parameters[Plugin::API_PARAMETER_ATTRIBUTES];
+            foreach ($attributes as $key => $value) {
+                if ($key === 'cat' && $category) {
+                    continue;
+                }
+
+                $request->setAttributeParam($key, $value);
+            }
+        }
+
+        if ($category && ($categoryFullName = $this->getCategoryName($category))) {
+            $request->setParam('selected', ['cat' => [$categoryFullName]]);
+            //TODO: remove after testing
+            $this->logger->error('Category full name: ' . $categoryFullName);
+        }
+
+        if (isset($parameters[Plugin::API_PARAMETER_SORT_ORDER]) && in_array($parameters[Plugin::API_PARAMETER_SORT_ORDER], Plugin::API_SORT_ORDER_AVAILABLE_OPTIONS)) {
```

Comment:
This won't work since the constant's value is `order` but Plentymarkets uses `sorting`. This was actually fixed before, seems like it happend during resolving some merge conflicts.